schemaless/create_schemaless.py

```diff
@@ -136,26 +173,38 @@ def dump_and_diff(sources, outfile, schemaless_file, the_date=None):
         mapblklot_gen.init(args.parcel_data_file)
 
     sources = []
-    if args.planning_file:
-        sources.append(Planning(args.planning_file))
-    if args.pts_file:
-        sources.append(PTS(args.pts_file))
-    if args.tco_file:
-        sources.append(TCO(args.tco_file))
-    if args.mohcd_pipeline_file:
-        sources.append(MOHCDPipeline(args.mohcd_pipeline_file))
-    if args.mohcd_inclusionary_file:
-        sources.append(MOHCDInclusionary(args.mohcd_inclusionary_file))
-    if args.permit_addenda_file:
-        sources.append(PermitAddendaSummary(args.permit_addenda_file))
-    if args.affordable_file:
-        sources.append(AffordableRentalPortfolio(args.affordable_file))
-    if args.oewd_permits_file:
-        sources.append(OEWDPermits(args.oewd_permits_file))
+    dl_sources = {}
+    destdir = tempfile.mkdtemp()
+
+    with futures.ThreadPoolExecutor(
+            thread_name_prefix="schemaless-download") as executor:
+        for (source, arg) in [
+                (Planning, args.planning_file),
+                (PTS, args.pts_file),
+                (TCO, args.tco_file),
+                (MOHCDPipeline, args.mohcd_pipeline_file),
+                (MOHCDInclusionary, args.mohcd_inclusionary_file),
+                (PermitAddendaSummary, args.permit_addenda_file),
+                (AffordableRentalPortfolio, args.affordable_file),
+                (OEWDPermits, args.oewd_permits_file)]:
+            if arg:
```

Comment:
How come we don't check the no download value first? if someone mistakenly put no_download to False but passed in args (I guess a contrived example) would this handle it?

Reply:
`no_download False` means it'll download anything not explicitly passed in, and it will print logs to that effect. Anything manually passed in is always used, anything not passed in is downloaded unless `--no_download True` is specified. It will only error out if `--no_download True` and no other flags are given.

Reply:
Ahhh that's what was confusing, I thought that no download meant all arguments would always be ignored, i guess we opt to prefer a manually passed in source argument always which is fine